ssc/sscapi.cpp

```diff
@@ -50,6 +50,7 @@
 #include <stdio.h>
 #include <cstring>
 #include <iostream>
+#include <vector>
```

Comment:
Please verify and test all code generation in SAM for all languages before finalizing the sscapi changes - the sscapi.h is generated for all code generation internally and may need to be updated if these changes are included.

Reply:
I've only added new functions to sscapi (for ssc_var_t and SSC_DATARR & SSC_DATMAT), and the code generator will not be using any of those since no SAM variable will be of those new types.

Reply:
going to resolve all comments about sscapi + GUI so we can discuss from a high level